src/platform-implementation-js/dom-driver/gmail/views/gmail-thread-row-view.ts

```diff
@@ -463,52 +470,74 @@ class GmailThreadRowView {
 
           imageMod = null;
         }
-      } else {
-        if (!imageMod) {
-          imageMod = this._modifications.image.unclaimed.shift();
 
-          if (!imageMod) {
-            imageMod = {
-              iconSettings: {},
-              iconWrapper: document.createElement('div'),
+        return;
+      }
 
-              remove() {
-                this.iconWrapper.remove();
-              },
-            };
-            imageMod.iconWrapper.className =
-              'inboxsdk__thread_row_icon_wrapper';
-          }
+      if (!imageMod) {
+        imageMod = this._modifications.image.unclaimed.shift();
+
+        if (!imageMod) {
+          imageMod = {
+            iconSettings: {},
+            iconWrapper: document.createElement('div'),
 
-          this._modifications.image.claimed.push(imageMod);
+            remove() {
+              this.iconWrapper.remove();
+            },
+          };
+          imageMod.iconWrapper.className = 'inboxsdk__thread_row_icon_wrapper';
+          if (iconDescriptor.image) {
+            imageMod.iconWrapper.classList.add(s.minWidthIconWrapper);
+          }
         }
 
-        const { iconSettings, iconWrapper } = imageMod;
-        updateIcon(
-          iconSettings,
-          iconWrapper,
-          false,
-          iconDescriptor.imageClass,
-          iconDescriptor.imageUrl,
+        this._modifications.image.claimed.push(imageMod);
+      }
+
+      const { iconSettings, iconWrapper } = imageMod;
+
+      updateIcon(
+        iconSettings,
+        iconWrapper,
+        false,
+        iconDescriptor.imageClass,
+        iconDescriptor.imageUrl,
+      );
+
+      if (iconDescriptor.image) {
+        this.#imageContainerRefreshResolution?.();
+
+        const promise = new Promise<void>(
+          (res) => (this.#imageContainerRefreshResolution = res),
         );
-        const containerRow =
-          this._elements.length === 3 ? this._elements[2] : this._elements[0];
-        containerRow.classList.add('inboxsdk__thread_row_image_added');
 
-        if (iconDescriptor.tooltip) {
-          iconWrapper.setAttribute('data-tooltip', iconDescriptor.tooltip);
-        } else {
-          iconWrapper.removeAttribute('data-tooltip');
-        }
+        const child = iconWrapper.appendChild(document.createElement('div'));
```

Comment:
should the `child` be created on each stream event? Looks like in this code a lot of checks are added to prevent duplicated creation of iconWrapper, so i think for the image element it should also only created it during imageMod creation for example.

Reply:
This is indeed leaking memory. @Macil pointed out elsewhere that due to optimizations around SDK routing, we leave elements in the DOM.